File: two_sum/two_sum.py
def two_sum(nums, target):
    seen = {}
    for i, num in enumerate(nums):
        diff = target - num
        if diff in seen:
            return [seen[diff], i]
        seen[num] = i
    raise ValueError("No two sum solution found.")

nums = [2, 7, 11, 15]
target = 9
result = two_sum(nums, target)
print("Indices of the two numbers that add up to the target:", result)  # Output: [0, 1]

File: two_sum/two_sum.ts
export {};

function twoSum(numbers: number[], target: number): number[] {
  const numMap = new Map<number, number>();
  for (let i = 0; i < numbers.length; i++) {
    const diff = target - numbers[i];
    if (numMap.has(diff)) {
      return [numMap.get(diff)!, i];
    }
    numMap.set(numbers[i], i);
  }
  throw new Error("No two sum solution found");
}

const numbers: number[] = [2, 7, 11, 15];
const target: number = 9;
const result: number[] = twoSum(numbers, target);
console.log(`Indices of the two numbers that add up to ${target}:`, result);
